toolchain/lower/file_context.cpp

```diff
@@ -255,8 +265,12 @@ auto FileContext::BuildFunctionDecl(SemIR::FunctionId function_id,
     if (!param_pattern.runtime_index.has_value()) {
       continue;
     }
-    auto param_type_id =
-        SemIR::GetTypeInSpecific(sem_ir(), specific_id, param_pattern.type_id);
+    auto param_type_id = param_pattern.type_id.AsConstantId().is_symbolic()
+                             ? SemIR::GetTypeInSpecific(sem_ir(), specific_id,
+                                                        param_pattern.type_id)
+                             : param_pattern.type_id;
```

Comment:
This looks sort of like the smaller part of the above lambda, maybe worth factoring out/sharing.